### DISCUSSION TITLE
Is Log4j2 compatible with edtftpj-pro.jar?

### DISCUSSION THREAD

**Discussion by TziporaG** [apache/logging-log4j2#2488] [Q&A] [answered]
I have a java application that utilizes ftp file transfers using the edtftpj-pro.jar library. When I intergrated log4j2 into my codebase, new errors were introduced making me wonder if these libraries are compatible. I have installed the log4j-1.2-api-2.23.jar as an attempt to "bridge" any gaps. Please advise.
Thank you

**Comment by ppkarwasz** [comment]
@TziporaG,

What errors are showing up in your logs?

**Reply by TziporaG** [reply]
@ppkarwasz 
Here is an example. This exception is not thrown with the previous log4j set up. The otheer new exception that is thrown is also related to the com.enterprisedt.net package. Thanks

#################################### SFTP EXCEPTION LOG BEGINS ######################################## 
com.enterprisedt.net.ftp.ssh.SSHFTPException: 
The SSH client has not yet connected to the server.  
The requested action cannot be performed until after a connection has been established.
com.enterprisedt.net.ftp.ssh.SCPClient.checkConnection(SCPClient.java:1233)
com.enterprisedt.net.ftp.ssh.SSHFTPClient.chdir(SSHFTPClient.java:1681)
com.dci.docubuilder.ftp.className.--(--)
com.dci.docubuilder.ftp.className.main(--)
com.dci.docubuilder.ftp.className.--(--) 
#################################### SFTP EXCEPTION LOG ENDS ########################################

**Reply by TziporaG** [reply, marked as answer]
Update- its working now, the issue was not related to the logger in the end. Thanks
